api/provider/aws/environment_update.go

```diff
@@ -33,6 +39,87 @@ func (e *EnvironmentProvider) Update(req models.UpdateEnvironmentRequest) error
 		}
 	}
 
+	if req.Links != nil {
+		addRemoveLinks := map[string]bool{}
+		secGrpIDtoEnvID := map[string]string{}
+
+		environmentSG, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqEnvironmentID))
+		if err != nil {
+			return err
+		}
+
+		// check which environments are currently linked
+		for _, ipPermission := range environmentSG.IpPermissions {
```

Comment:
If we rely on the just the tags however, it could lead to a situation where the actual state can drift from our tags store. We need at least a way to verify the current state is as is right? If so I would have thought we can rely on making the assumptions being made here as we appending custom rules to security groups linking other security groups created via l0 is something that we expose (apart from the links).

Reply:
There's pros and cons to each, for sure. I just think it would be too costly/complex to verify/map an environment's security group ingress rules to links vs. just going with tags. If it's any consolation, we are relying on tags as the source of truth for task and deploy listing (via the `arn` tags)

Reply:
ok - will update it to rely on tags instead. ✅